rust/parsing.rs

```diff
@@ -0,0 +1,945 @@
+                        let mut i: u8 = 0;
+
+                        while i < 6 {
+                            if self.current >= '0' && self.current <= '9' {
+                                datetime.microsecond =
+                                    datetime.microsecond * 10 + self.current.to_digit(10).unwrap();
+                            } else if i == 0 {
+                                // One digit minimum is required
+                                return Err(self.unexpected_character_error("subsecond", 1));
+                            } else {
+                                break;
+                            }
+
+                            self.inc();
+                            i += 1;
+                        }
+
+                        // Drop extraneous digits
+                        while self.current >= '0' && self.current <= '9' {
+                            self.inc();
+                        }
+
+                        // Expand missing microsecond
+                        while i < 6 {
+                            datetime.microsecond *= 10;
+                            i += 1;
+                        }
+                    }
+
+                    if !datetime.extended_date_format {
+                        return Err(self.parse_error("Cannot combine \"basic\" date format with \"extended\" time format (Should be either `YYYY-MM-DDThh:mm:ss` or `YYYYMMDDThhmmss`).".to_string()));
+                    }
+                }
+            } else {
+                // Minute and second in compact format (mmss)
+
+                // Minute
+                datetime.minute = self.parse_integer(2, "minute")?;
+
+                if !self.end() && self.current != 'Z' && self.current != '+' && self.current != '-'
+                {
+                    // Optional second
+
+                    datetime.second = self.parse_integer(2, "second")?;
+
+                    if self.current == '.' || self.current == ',' {
+                        // Optional fractional second
+                        self.inc();
+
+                        datetime.microsecond = 0;
+                        let mut i: u8 = 0;
+
+                        while i < 6 {
+                            if self.current >= '0' && self.current <= '9' {
+                                datetime.microsecond =
+                                    datetime.microsecond * 10 + self.current.to_digit(10).unwrap();
+                            } else if i == 0 {
+                                // One digit minimum is required
+                                return Err(self.unexpected_character_error("subsecond", 1));
+                            } else {
+                                break;
+                            }
+
+                            self.inc();
+                            i += 1;
+                        }
+
+                        // Drop extraneous digits
+                        while self.current >= '0' && self.current <= '9' {
```

Comment:
std has a helper method for this
```suggestion
                        while self.current.is_ascii_digit() {
```